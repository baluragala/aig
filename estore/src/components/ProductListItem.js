import React, { Component } from "react";

class ProductListItem extends Component {
  render() {
    const { title, stock, price } = this.props.product;
    return (
      <div>
        <h2>
          {title}({stock})
        </h2>
        <h4>{price}</h4>
      </div>
    );
  }
}

export default ProductListItem;
