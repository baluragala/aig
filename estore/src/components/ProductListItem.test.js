import ProductListItem from "./ProductListItem";
import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Link } from "react-router-dom";
import { Map } from "immutable";
import { stringify } from "query-string";
import { shallowToJson } from "enzyme-to-json";

Enzyme.configure({
  adapter: new Adapter()
});

describe("ProductListItem", () => {
  it("should render product information", () => {
    const product = Map({
      id: 1,
      title: "p1",
      price: 100,
      stock: 20
    });

    const domWrapper = shallow(<ProductListItem product={product} />);

    expect(domWrapper.find("h2").length).toEqual(1);
    expect(domWrapper.find(Link).length).toEqual(1);
    let children = domWrapper.find(Link).children();
    expect(children.contains("p1")).toBeTruthy();

    expect(domWrapper.find(Link).prop("to")).toEqual(
      `/products/${product.get("id")}?${stringify({ related: true })}`
    );
  });

  it("should handle click event", () => {
    const product = Map({
      id: 1,
      title: "p1",
      price: 100,
      stock: 20
    });

    const mockedClickHandler = jest.fn();
    const domWrapper = shallow(
      <ProductListItem product={product} onSale={mockedClickHandler} />
    );
    domWrapper
      .find("button")
      .first()
      .simulate("click");
    expect(mockedClickHandler).toHaveBeenCalledTimes(1);
    expect(mockedClickHandler).toHaveBeenCalledWith(1);
  });

  it("should pass snapshot", () => {
    const product = Map({
      id: 1,
      title: "p1",
      price: 100,
      stock: 20
    });

    const domWrapper = shallow(<ProductListItem product={product} />);
    expect(shallowToJson(domWrapper)).toMatchSnapshot();
  });
});
