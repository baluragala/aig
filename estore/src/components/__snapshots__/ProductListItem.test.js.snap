// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListItem should pass snapshot 1`] = `
<div>
  <h2>
    <Link
      replace={false}
      to="/products/1?related=true"
    >
      p1
      20
    </Link>
  </h2>
  <h4>
    100
  </h4>
  <button
    onClick={[Function]}
  >
    Sell
  </button>
</div>
`;
